File: Offer/C++/offer52.cc
#include<iostream>
using namespace std;

struct ListNode{
    int val;
    ListNode *next;
    ListNode(int x):val(x),next(NULL){}
};

class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        if(headA == nullptr || headB == nullptr){
            return nullptr;
        }
        ListNode* Pa = headA;
        ListNode* Pb = headB;

        while(Pa != Pb){
            Pa = Pa == nullptr ? headB:Pa->next;
            Pb = Pb == nullptr ? headA:Pb->next;
        }
        return Pa;
    }
};

File: Offer/src/linkedList/Offer52.java
package linkedList;

public class Offer52 {
    ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode p1 = headA,p2 = headB;
        while(p1 != null && p2 != null){
            if(p1.val != p2.val){
                p1 = p1.next;
                p2 = p2.next;
            }
        }
        return p1;
    }
}
